pages/api/recipes/[id].ts

```diff
@@ -28,8 +53,25 @@ export default async function handler(
         response.status(400).json({ error: 'an unknown error occured' });
       }
     }
-  } else if (request.method === 'DELETE') {
+  }
+
+  if (request.method === 'DELETE') {
+    // Check if Session
+    if (!session) {
+      return response.status(401).json({ message: 'Unauthorized' });
+    }
+    // Get the recipe by ID and check if the recipe Owner is equal to the Session User Id
     try {
+      const existingRecipe = await Recipe.findById(id);
+
+      if (!existingRecipe) {
+        return response.status(404).json({ message: 'Recipe not found' });
+      }
+
+      if (!existingRecipe.authorId || !session?.user?.id) {
+        return response.status(400).json({ message: 'Invalid Request' });
+      }
+      // Delete Recipe
```

Comment:
You forgot to check whether the user making the request is the user who owns this recipe. Have a look @ lines 42 - 44.

Reply:
Thanks! I will have a look!